src/App.tsx

```diff
@@ -5,7 +5,15 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (movie: Movie | null) => {
+    if (!movie) {
+      return;
+    }
+
+    setMovies([...movies, movie]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setMovies
```
setMovies((movies) => [...movies, movie]);
```